tools/generator/tests/test_AnDOGenerator.py

```diff
@@ -79,17 +80,16 @@ def test_complex_parameters(self):
         self.assertEqual(self.guid, ses.guid)
         self.assertEqual(self.sesID, str(ses.sesID))
 
-    # TODO: This check is still failing, review engine/check_Path for consistency
+
     def test_paths(self):
-        ses = AnDOSession(self.expName, self.guid, self.sesID)
+        ses = AnDOSession(self.expName, self.guid, self.sesID, self.ephys)
 
-        expected = os.path.join(f'exp-{self.expName}', f'sub-{self.guid}', f'ses-{self.sesID}')
+        expected = os.path.join(f'exp-{self.expName}', f'sub-{self.guid}', f'ses-{self.sesID}',f'{self.ephys}')
         self.assertEqual(expected, ses.get_session_path())
-
-        self.assertEqual(3, len(ses.get_all_folder_paths()))
+        self.assertEqual(1, len(ses.get_all_folder_paths())) # There is now only one subfolder called ephys
 
     def test_generate_folders(self):
-        ses = AnDOSession(self.expName, self.guid, self.sesID)
+        ses = AnDOSession(self.expName, self.guid, self.sesID,self.ephys)
```

Comment:
This additional argument is not implemented as far as I can see and it also does not really make sense, since the 'ephy' part is fixed for now in the new BEP....

Reply:
Reverted

Reply:
but we still need to add it at the end like :
``` ses = AnDOSession(self.expName, self.guid, self.sesID,"ephys")```
Right ?

Reply:
Yes we can, but I wouldn't since it's not an actual variable parameter, but fixed for us. Maybe it would make sense to define the string 'ephys' as a kind of global variable within the AnDO generator?

Reply:
I'm not sure about a global variable, since we will soon want to extend this to other modalities like "vsdi"... so we'd need a solution that could handle both...

although I understand that the checker will not work exactly the same way for ephys and vsdi (since the required metadata might not be the same)... I guess we probably need to have a mechanism that checks for different things according to the modality, no? but we don't have to implement this mechanism right now... just think about it for this specific case of handling the 'ephys' string is probably enough for now!!

Reply:
...or maybe a global variable for "ephys" is the best / most simple solution for now! and we'll see how to extend this to "vsdi" later... ;)

Reply:
Ok so what do you think if we add in the AnDO generator : 
```python
MANDATOORY_SUBFOLDER = ["ephys"]
```

then in : 

```python
def get_session_path(self):
        path = os.path.join(f'exp-{self.expName}',
                            f'sub-{self.guid}',
                            f'ses-{self.sesID}',
                            folder for folder in MANDATOORY_SUBFOLDER
                            )
```
This should work right ?

Reply:
no need for it to be a list... just use
`MANDATORY_SUBFOLDER = "ephys"`
(with only one O at mandatOOry ;) )

WDYT @JuliaSprenger

Reply:
Sounds good to me!